keras/backend/tensorflow/nn.py

```diff
@@ -787,3 +787,69 @@ def batch_normalization(
         scale=scale,
         variance_epsilon=epsilon,
     )
+
+
+def ctc_labels_to_sparse(labels, label_length):
+    """Converts CTC labels to a sparse tensor.
+
+    Arguments:
+        labels: Tensor `(batch_size, max_label_length)` containing
+            the dense CTC labels.
+        label_length: Tensor `(batch_size,)` containing the
+            sequence length for each label sequence in the batch.
+
+    Returns:
+        Sparse tensor representation of `labels`.
+    """
+    max_label_len = tf.shape(labels)[1]
+
+    mask = tf.sequence_mask(label_length, max_label_len)
+    indices = tf.where(mask)
+    values = tf.boolean_mask(labels, mask)
+
+    return tf.SparseTensor(
+        indices=indices,
+        values=values,
+        dense_shape=tf.cast(tf.shape(labels), dtype="int64"),
+    )
+
+
+def ctc_batch_cost(
+    target,
+    output,
+    target_length,
+    output_length,
+    blank_index=0,
+):
+    """Runs CTC (Connectionist Temporal Classification) loss on each batch element.
+
+    Arguments:
+        target: Tensor `(batch_size, max_target_length)` containing the
+            target sequences.
+        output: Tensor `(batch_size, max_output_length, label_classes)`
+            containing the output of the softmax.
+        target_length: Tensor `(batch_size,)` containing the sequence length
+            for each target sequence in the batch.
+        output_length: Tensor `(batch_size,)` containing the sequence length
+            for each output sequence in the batch.
+        blank_index: The value in `target` and `output` that represents the
```

Comment:
Is there a name that would be more consistent with naming conventions elsewhere in the API, like `padding_index`?

Reply:
Blanks can actually occur in between two "actual" labels and are not necessarily padding, so it might not fit the CTC spec (which explicitly calls these blanks). Should we stick to the CTC name? It seems quite specific.

Reply:
Sure, that is fine!

Reply:
Sure, this is done!